[IMP] Port TMS report to new API

seb-elico opened the pull request:
Port TMS report to new API

seb-elico committed:
[WIP] Port TMS report to new API

seb-elico committed:
Change module version

seb-elico committed:
Add FIXME and improve string separator

seb-elico committed:
Add missing space in help message

seb-elico committed:
Remove parent query since ID is unique already.

This "trick" had been introduced for the project_completion_report
module which actually needs it since it returns different types of
objects (e.g. task/issue/TMS) which can have the same ID. Here,
it's not the case since only 1 type of object (e.g. TMS) is
returned.

seb-elico commented:
@victormartinelicocorp @Wizr @Zhouhao12345 can you answer the FIXMEs in my code please?

elicoidal commented on timesheet_activity_report/__openerp__.py:
```diff
@@ -3,7 +3,7 @@
 {
     'name': 'Timesheet Activities Report',
-    'version': '8.0.1.0.3',
+    'version': '8.0.1.1.0', #FIXME is it the right version? I assume the module needs to be upgraded...
```
correct: restart+upgrade, no migration (https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers)

seb-elico replied:
@elicoidal Actually, so far the fields are the same, it's just a port to new API. So I'm wondering if an upgrade of the module is necessary!

elicoidal replied:
It depends if you include the change in the SQL view or not in this PR.

seb-elico replied:
Indeed I simplified the SQL view (not for the BR yet, I wanted to do that in a separate PR), you're right. Thanks!

elicoidal reviewed (approved):
LGTM (no feedback on the new API questions though)

seb-elico committed:
Remove FIXMEs, add comments and rollback name of init()

seb-elico committed:
Remove default pivot table fields to improve performance

When browsing thousands of TMS linked to hundreds of projects and
switching to the graph view, the default fields made the loading
time of the view way too long

seb-elico committed:
Change parent class to standard Model

seb-elico committed:
In new API, when you override init() you still need the cr

Example:
https://github.com/OCA/account-financial-tools/blob/8.0/account_reconcile_trace/models/account_reconcile_trace.py#L59

seb-elico committed:
Noah forgot to remove this comment in #125

seb-elico committed:
Put back task category (removed by #147)

seb-elico committed:
Improve readability of XML file

seb-elico committed:
Add dependency for task category

seb-elico committed:
Fix README for Travis + latest template

seb-elico committed:
Add support key for Travis LINT check

seb-elico committed:
Fix Travis

seb-elico committed:
Adapt TMS report to new link between BR and project

seb-elico commented:
@Elico-Corp/odoo-reviewers Final version, please review

elicoidal commented on timesheet_activity_report/README.rst:
```diff
@@ -15,37 +15,48 @@ Installation
 
- * have basic modules installed (hr_timesheet, project_timesheet, project_issue_sheet)
-
+#. Have basic modules installed (hr_timesheet, project_timesheet, project_issue_sheet)
```
doesnot make sense to make an enumerated list with one item no?

seb-elico replied:
I just copy/pasted the README from another module that also had only 1 item. Any recommendation then?

martintorresvm replied:
IMO this looks cleaner:
```
Have basic modules installed:

   * hr_timesheet
   * project_timesheet
   * project_issue_sheet
```

elicoidal commented on timesheet_activity_report/README.rst:
```diff
@@ -15,37 +15,48 @@ Installation
 Bug Tracker
 ===========
 
-Bugs are tracked on `GitHub Issues <https://github.com/Elico-Corp/odoo/issues>`_.
```
I miss the runbot

seb-elico replied:
Project is not under Runbot

elicoidal replied:
OK

elicoidal commented on timesheet_activity_report/report/timesheet_activity_report.py:
```diff
@@ -2,125 +2,120 @@
+            ('timesheet', 'Timesheet'),
+        ], 'Type', readonly=True,
+        help="Type is used to separate Tasks, Issues and "
+             "Timesheets input directly")
```
Timesheet direct input

seb-elico replied:
@elicoidal Actually, the full sentence should be:

> "Type is used to separate Timesheets input in a Task, in an Issue or in Timesheet Activities" 

Is that ok for you?

elicoidal replied:
Another option (and name could be source or origin maybe but n-t-h)
Type is based on the origin of the input (Task, Issue or Timesheet Activity Report)

elicoidal reviewed (changes requested):
some details. overall LGTM but needs to be tested

elicoidal reviewed (changes requested):
some nice to have details

seb-elico commented:
@victormartinelicocorp Any idea why Travis is failing?

````
File "/home/travis/build/Elico-Corp/odoo-addons/timesheet_activity_report/report/timesheet_activity_report.py", line 121, in init
)""")
File "/home/travis/odoo-8.0/openerp/sql_db.py", line 171, in wrapper
return f(self, *args, **kwargs)
File "/home/travis/odoo-8.0/openerp/sql_db.py", line 247, in execute
res = self._obj.execute(query, params)
ProgrammingError: column p.business_requirement_id does not exist
LINE 58:                         ON b.id = p.business_requirement_id
````

seb-elico committed:
BR depend from OCA rather than Noah's fork

Also take the opportunity to clean the file to make it compatible
for all branches

seb-elico committed:
Change help message based on CR feedback

seb-elico commented:
@victormartinelicocorp I found why it's failing, I just pushed a new `oca_dependencies.txt`. Let's see if it'll fix the bug

seb-elico committed:
Fix README based on CR feedback

seb-elico committed:
BR module is already included by BR Deliverable Project

seb-elico commented:
@elicoidal @victormartinelicocorp can you approve please? Let's merge as soon as Travis build is green
